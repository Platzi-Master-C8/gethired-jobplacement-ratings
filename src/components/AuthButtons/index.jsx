import React from 'react';

import Button from '@mui/material/Button';

import { goToEnterprise } from '../../utils';

export const AuthenticationButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Log in
        </Button>
    );
};

export const SignupButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Sign up
        </Button>
    );
};
