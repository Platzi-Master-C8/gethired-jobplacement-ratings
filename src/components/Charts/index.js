import RadarChart from "./RadarChart";
import LineChart from "./LineChart";
export { RadarChart, LineChart }