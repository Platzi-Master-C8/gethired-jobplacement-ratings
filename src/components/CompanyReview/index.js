import CompanyReview from './CompanyReview';

export { CompanyReview };
