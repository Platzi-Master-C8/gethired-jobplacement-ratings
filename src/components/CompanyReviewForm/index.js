import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
