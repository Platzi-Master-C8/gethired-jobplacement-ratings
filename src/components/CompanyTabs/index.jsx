import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
