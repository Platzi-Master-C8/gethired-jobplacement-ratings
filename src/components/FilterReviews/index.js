import FilterReviews from './FilterReviews';

export {
    FilterReviews
}