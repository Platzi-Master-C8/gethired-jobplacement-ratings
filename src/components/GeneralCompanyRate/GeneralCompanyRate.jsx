import React from 'react';
import './GeneralCompanyRate.scss';

const GeneralCompanyRate = () => (
    <div className="generalCompanyRate">
        <img alt="company-logo" src="" />
        <h2>Company Name</h2>
        <h6>5.0</h6>
        <i className="fa fa-start" />
        <i className="fa fa-start" />
        <i className="fa fa-start" />
        <i className="fa fa-start" />
        <i className="fa fa-start" />
        <span>123 Reviews</span>
    </div>
);

export default GeneralCompanyRate;
