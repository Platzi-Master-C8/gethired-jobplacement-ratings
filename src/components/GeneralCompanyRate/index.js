import GeneralCompanyRate from './GeneralCompanyRate';

export { GeneralCompanyRate };
