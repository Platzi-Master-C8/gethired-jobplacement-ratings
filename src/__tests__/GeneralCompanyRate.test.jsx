import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import { GeneralCompanyRate } from '../components/GeneralCompanyRate';

describe('<CompanyTabs />', () => {
    test('Component render', () => {
        const { getByText } = render(<GeneralCompanyRate />);

        expect(getByText(/reviews/i)).toBeTruthy();
        expect(getByText(/write a review/i)).toBeTruthy();
    });

    test('Get components', () => {
        const { getByText, getByAltText } = render(<GeneralCompanyRate />);

        expect(getByAltText('green iguana')).toBeTruthy();
        expect(getByText(/company/i)).toBeTruthy();
        expect(getByText(/write a review/i)).toBeTruthy();
    });

    test('Open review modal', () => {
      const { getByText } = render(<GeneralCompanyRate />);

      fireEvent.click(getByText(/write a review/i));
      expect(getByText(/review the company/i)).toBeTruthy();
  });
});
