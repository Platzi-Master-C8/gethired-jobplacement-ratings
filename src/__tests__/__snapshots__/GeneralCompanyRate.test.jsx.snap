// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneralCompanyRate /> Component render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1qi2013-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-3 css-sktcqb-MuiGrid-root"
      >
        <img
          alt="green iguana"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1y7obv6-MuiCardMedia-root"
          height="140"
          src="https://images.pexels.com/photos/430205/pexels-photo-430205.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-5 css-15go1ru-MuiGrid-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
        >
          Company
        </h1>
        <div
          class="MuiBox-root css-kblalv"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            4.2
          </h6>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarRateIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.43 10 12 2l-2.43 8H2l6.18 4.41L5.83 22 12 17.31 18.18 22l-2.35-7.59L22 10z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarRateIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.43 10 12 2l-2.43 8H2l6.18 4.41L5.83 22 12 17.31 18.18 22l-2.35-7.59L22 10z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarRateIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.43 10 12 2l-2.43 8H2l6.18 4.41L5.83 22 12 17.31 18.18 22l-2.35-7.59L22 10z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarRateIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.43 10 12 2l-2.43 8H2l6.18 4.41L5.83 22 12 17.31 18.18 22l-2.35-7.59L22 10z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarHalfIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4V6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            123 reviews
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-1imo3bq-MuiGrid-root"
      >
        <div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-oav0mg-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Write a Review
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
